include/ArchiveEnumerations.h

```diff
@@ -0,0 +1,124 @@
+    - Software Patents You Do Not Wish to Freely License
+    - Any Kind of Linking to Non-GPL licensed Works
+    - Are Currently In Violation of Another Copyright Holder's GPL License
+    - If You want to change our code and not add a few hundred MB of stuff to
+        your distribution
+
+   These and other limitations could cause serious legal problems if you ignore
+   them, so it is best to simply contact us or the Free Software Foundation, if
+   you have any questions.
+
+   Joseph Toppi - toppij@gmail.com
+   John Blackwood - makoenergy02@gmail.com
+*/
+#ifndef Mezz_IOStreams_ArchiveEnumerations_h
+#define Mezz_IOStreams_ArchiveEnumerations_h
+
+#ifndef SWIG
+    #include "BitFieldTools.h"
+#endif
+
+namespace Mezzanine
+{
+    /// @brief Used to indicate what kind of resources the Entresol should look for
+    enum class ArchiveType
+    {
+        Invalid    = 0,  ///< Uninitialized or Indicates an error condition of some kind.
+        FileSystem,      ///< No archive format, raw binary on disk.
+        Zip,             ///< Archive abiding by the PKWARE Zip format.
+        SevenZ           ///< Archive abiding by the 7zip format.
+    };
+
+    /// @brief Used to indicate an algorithm of compression.
+    enum class CompressionMethod
+    {
+        Unknown   = 0,   ///< Uninitialized or Indicates an error condition of some kind.
+        None,            ///< No compression is applied.
+        Deflate,         ///< Common DEFLATE compression.
+        Deflate64,       ///< A 64-bit variant of Deflate.
+        BZip2,           ///< A compression method that is slow to compress, but fast to decompress.
+        ZPAQ,            ///< A compression method that uses a combination of LZ77 and BWT.
+        Brotli,          ///< A google compression method based on LZ77 offering denser compression than DEFLATE.
+        LZMA,            ///< The compression method used by 7zip offering dense compression.
+        LZ4              ///< A compression method focused on compression and decompression speed.
+    };
+
+    /// @brief Used to indicate an algorithm of encryption.
+    enum class EncryptionMethod
+    {
+        Unknown  = 0,    ///< Uninitialized or Indicates an error condition of some kind.
+        None,            ///< No encryption is applied.
+        PKWARE,          ///< Proprietary encryption used by PKWARE in Zip/Zip64 files.
+        AES_128,         ///< A symmetric encryption with a 128-bit key size.
+        AES_192,         ///< A symmetric encryption with a 192-bit key size.
+        AES_256,         ///< A symmetric encryption with a 256-bit key size.
+        RSA_128,         ///< An asymmetric encryption with a 128-bit key size.
+        RSA_192,         ///< An asymmetric encryption with a 192-bit key size.
+        RSA_256,         ///< An asymmetric encryption with a 256-bit key size.
+        TDES_112,        ///< Also 3TES. A symmetric encryption with a 112-bit key size.
```

Comment:
TripleDES is the most common way I see it spelled, but it is old and bad. I am surprised to see any support for it in the Mezzanine.

Reply:
I just looked through some libraries out in the wild to see what we could feasibly support.  This PR, and every PR for the forseeable future won't include support for it.  For that reason I could just up and remove it from the enum.

Reply:
It is too weak to support, I think removing is the safest option.

Reply:
It's been removed.